src/xmppjs/XJSInstance.ts

```diff
@@ -430,13 +438,59 @@ export class XmppJsInstance extends EventEmitter implements IBifrostInstance {
         });
     }
 
+    private async getMucAvatar(room: string): Promise<Element> {
+        const id = uuid();
+        // check if MUC supports avatars
+        const result = await this.sendIq(new StzaIqDiscoInfo(this.myAddress.toString(), room, uuid(), "get"));
+        const supportVCards = result.getChild("query")?.getChildByAttr("var", "vcard-temp");
+        if (!supportVCards) {
+            throw Error("MUC doesn't support avatars");
+        }
+        const res = new Promise((resolve: (e: Element) => void, reject) => {
+            const timeout = setTimeout(() => reject(Error("Timeout")), 5000);
+            this.once(`iq.${id}`, (stanza: Element) => {
+                clearTimeout(timeout);
+                const vCard = (stanza.getChild("vCard") as unknown as Element);
+                if (vCard) {
+                    resolve(vCard);
+                }
```

Comment:
These type casts shouldn't  be necessary – this should already be `Element|undefined`.

```suggestion
                const vCard = stanza.getChild("vCard");
                if (vCard) {
                    resolve(vCard!);
                }
```

Reply:
No problem on the change, I just left it as it previously was.

Reply:
Actually removing the type casts cause the following when building:
```
src/xmppjs/XJSInstance.ts(498,29): error TS2345: Argument of type 'import("/node_modules/@types/ltx/lib/Element").Element' is not assignable to parameter of type 'import("/node_modules/@types/xmpp__xml/index").Element'.
  Type 'Element' is missing the following properties from type 'Element': append, prepend
```

So I'll revert the change for now.

Reply:
@tadzik could you review the type cast issue above? As your suggested change causes the problem above.